packages/astro/src/core/cookies/cookies.ts

```diff
@@ -2,22 +2,16 @@ import type { CookieSerializeOptions } from 'cookie';
 import { parse, serialize } from 'cookie';
 import { AstroError, AstroErrorData } from '../errors/index.js';
 
-export interface AstroCookieSetOptions {
-	domain?: string;
-	expires?: Date;
-	httpOnly?: boolean;
-	maxAge?: number;
-	path?: string;
-	sameSite?: boolean | 'lax' | 'none' | 'strict';
-	secure?: boolean;
-	encode?: (value: string) => string;
-}
+export type AstroCookieSetOptions = Pick<
+	CookieSerializeOptions,
+	'domain' | 'path' | 'expires' | 'maxAge' | 'httpOnly' | 'sameSite' | 'secure' | 'encode'
+>;
```

Comment:
Partitioned should also be here. Support was added in `cookie@0.6`, which Astro has been using since 4.0.

Reply:
This would also change the types `AstroCookies.set` API. And I think this is what @matthewp meant by [his comment](https://github.com/withastro/astro/pull/10671#discussion_r1551744943) about being selective when it comes to cookie attribute, probably since `partitioned` is still experimental?

I personally agree with supporting it, but I believe this to be a separate conversation amongst the maintainers. 😅

Reply:
It already has been working, I am only suggesting aligning the types with the actual current behavior.

As for its status, `cookie` doesn't consider it to be experimental - not anymore than everything else at least, the package itself is pre-1.0.